packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -93,47 +73,47 @@ void main() {
   );
 
   setUp(() {
+    testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
+      MethodChannel(tokenReceivedEventChannel.name),
+      handler,
+    );
     plugin = TestAmplifyPushNotifications(
       serviceProviderClient: mockServiceProviderClient,
       backgroundProcessor: () async => {},
       dependencyManager: dependencyManager,
     );
-
-    testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
-      MethodChannel(tokenReceivedEventChannel.name),
-      handler,
-    );
   });
 
   tearDown(log.clear);
 
   group('Push Notifications config', () {
     setUp(() {
-      log.clear();
-      plugin = TestAmplifyPushNotifications(
-        serviceProviderClient: mockServiceProviderClient,
-        backgroundProcessor: () async => {},
-        dependencyManager: dependencyManager,
-      );
-
-      testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
-        MethodChannel(tokenReceivedEventChannel.name),
-        handler,
-      );
+      // Sending the token after a delay so that token.first in configure gets the token.
+      // This is only required for tests since StreamQueue used for buffered stream in impl
+      // listens to the channel before token.first call making it useless to emit event when listened.
+      // This delay is replicated throughout the tests so token.first can fire.
+      Future.delayed(const Duration(microseconds: 1), () async {
```

Comment:
I don't get what this means or why it's needed. This is the equivalent of `Future.delayed(Duration.zero, ...` meaning "run in the next event loop". It's not clear why this is needed since the test plugin is creating the `StreamQueue` at the same time as the normal plugin (in the constructor).

If the test plugin is behaving differently than the normal plugin, then this test is really testing the test plugin which I don't think adds much value. The test plugin should behave as close as possible to the original plugin and we should be mocking one thing at a time (in this case the event channel).

Reply:
It's not the test plugin as I am just returning the constructed object with super values. 

Initially I was emitting the token event when a stream listens in on the event channel as you can see [here](https://github.com/aws-amplify/amplify-flutter/pull/2824/files#diff-2d8b06b2122317db94204b3791335562662be1056b2229fc3f543e0e26a8030dL62).

StreamQueue listens to the given stream internally and so the event would emit even before token.first gets called. 

But that was when I was using token.first but since I changed it to use bufferedStream.peek this wait is not required. Thanks for catching it, will change it back to prev state which works just fine.